noir-projects/noir-contracts/contracts/app_subscription_contract/src/main.nr

```diff
@@ -2,18 +2,17 @@ mod subscription_note;
 mod dapp_payload;
 
 contract AppSubscription {
```

Comment:
In this contract I know perform the block number checks via the router contract.